code/leslie/python/lab12.py

```diff
@@ -0,0 +1,76 @@
+class ATM:
+    def __init__(self, balance = 0, interest_rate = 0.001): #initializer with default values
+        self.balance = balance #member variables
+        self.interest_rate = interest_rate
+        
+    def check_balance(self): #method
+        return self.balance # returns account balance
+    
+    def deposit(self, amount):
+        self.balance += amount #deposits given amount in account
+       
+        
+    def check_withdrawal(self, amount): 
+        if amount < self.balance: #if withdrawl won't put account in negative, return True
+            return True        
+        
+    def withdraw(self, amount): #withdraws amount from account and returns it
+        self.balance -= amount
+        return amount
+    
+    def calc_interest(self): #returns amount of interest calculated on account. 
+        interest = self.balance * self.interest_rate
+        return interest
+    
+    def print_transactions(self, list_of_transactions):
+        self.list_of_transactions = list_of_transactions
+        print(transactions)
```

Comment:
While this approach technically performs the action described by the lab requirements, it would be a better design if `list_of_transactions` was stored as an attribute of the ATM instance itself. I'd like you to try to implement it that way and re-submit.

Reply:
This work was done on another branch